test/e2e/watchcache/watchcache_enabled_test.go

```diff
@@ -0,0 +1,101 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchcache
+
+import (
+	"bufio"
+	"context"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubernetesclientset "k8s.io/client-go/kubernetes"
+
+	"github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest"
+	wildwestv1alpha1 "github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest/apis/wildwest/v1alpha1"
+	wildwestclientset "github.com/kcp-dev/kcp/test/e2e/fixtures/wildwest/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestWatchCacheEnabled(t *testing.T) {
+	t.Parallel()
+	server := framework.SharedKcpServer(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	org := framework.NewOrganizationFixture(t, server)
+	cluster := framework.NewWorkspaceFixture(t, server, org)
+	cfg := server.DefaultConfig(t)
+	kubeClusterClient, err := kubernetesclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err)
+
+	// create cowboys.wildwest.dev CRD and an object
+	apiExtensionsClients, err := apiextensionsclient.NewClusterForConfig(cfg)
+	require.NoError(t, err)
+	crdClient := apiExtensionsClients.Cluster(cluster).ApiextensionsV1().CustomResourceDefinitions()
+	wildwest.Create(t, crdClient, metav1.GroupResource{Group: "wildwest.dev", Resource: "cowboys"})
+	wildwestClusterClient, err := wildwestclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err)
+	_, err = wildwestClusterClient.Cluster(cluster).WildwestV1alpha1().Cowboys("default").Create(ctx, &wildwestv1alpha1.Cowboy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "efficientluke",
+		},
+		Spec: wildwestv1alpha1.CowboySpec{
+			Intent: "should be kept in memory",
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// get the CRD from the cache a few times
+	for i := 0; i < 10; i++ {
+		res, err := wildwestClusterClient.Cluster(cluster).WildwestV1alpha1().Cowboys("default").List(ctx, metav1.ListOptions{ResourceVersion: "0"})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(res.Items), "expected to get exactly one cowboy")
+	}
+
+	// cache hits are recorded as a metric
+	rsp := kubeClusterClient.RESTClient().Get().AbsPath("/metrics").Do(ctx)
+	raw, err := rsp.Raw()
+	require.NoError(t, err)
+	scanner := bufio.NewScanner(strings.NewReader(string(raw)))
+	var totalCacheHits, cowboysCacheHit int
+	for scanner.Scan() {
+		txt := scanner.Text()
+		if strings.Contains(txt, "apiserver_cache_list_total") {
+			totalCacheHits++
+			if strings.Contains(txt, `resource_prefix="/wildwest.dev/cowboys/customresources`) {
```

Comment:
oh, this means we need work in metrics. It does not distinguish between workspaces and not between CRDs and APIBindings, etc.

Reply:
Also note that this test is run in parallel with other e2e also using the cowboys. Can we use a custom API group here, e.g. with a uuid as group name?

Reply:
I think we should be able to distinguish by 
```
resource_prefix="/group/resource/idenity"}
resource_prefix="/group/resource/customresources"}
```

we could pick up a different `gr`

Reply:
does it make sense?

Reply:
actually, are ok if some other controller increases the metrics? Then we shouldn't care about unique GRs.